### PR TITLE
[ZEPPELIN-5393] Polish rest, socket package of zeppelin-server

### PR DESCRIPTION
**Pull request by cuspymd**
### What is this PR for?
- Add final keyword to variables which are not re-assigned
- Delete unnecessary and unused codes

### What type of PR is it?
[Refactoring]

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5393

### How should this be tested?
* CI

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by jongyoul** [comment]
Who can verify if some test failures are flaky or not? I've checked the changes and LGTM.

**Comment by zjffdu** [comment]
@Reamer It looks like the K8s UT fails again
```
PodPhaseWatcherTest.testPhase:69 null
```

**Comment by zjffdu** [comment]
@Reamer It looks like the K8s UT fails again
```
PodPhaseWatcherTest.testPhase:69 null
```
https://github.com/apache/zeppelin/pull/4126/checks?check_run_id=2686162242

**Comment by zjffdu** [comment]
@Reamer It looks like the K8s UT fails again
```
PodPhaseWatcherTest.testPhase:69 null
```
https://github.com/apache/zeppelin/pull/4126/checks?check_run_id=2686162242

**Comment by Reamer** [comment]
> @Reamer It looks like the K8s UT fails again
> 
> ```
> PodPhaseWatcherTest.testPhase:69 null
> ```
> 
> https://github.com/apache/zeppelin/pull/4126/checks?check_run_id=2686162242

Yes, this test seems to be a flapping one. I cannot reproduce this problem locally.

**Comment by Reamer** [comment]
@zjffdu  I have created PR #4129. Hopefully this will fix the flapping test.

**Comment by zjffdu** [comment]
@Reamer #4129 is merged, could you rebase this PR ?

**Comment by Reamer** [comment]
> @Reamer #4129 is merged, could you rebase this PR ?

This PR is from @cuspymd :smile: 
@cuspymd please rebase.

**Comment by zjffdu** [comment]
LGTM, thanks @cuspymd

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 1]
```diff
@@ -251,27 +250,27 @@ public void onOpen(NotebookSocket conn) {
...
-      Message messagereceived = deserializeMessage(msg);
-      if (messagereceived.op != OP.PING) {
-        LOG.debug("RECEIVE: " + messagereceived.op +
-            ", RECEIVE PRINCIPAL: " + messagereceived.principal +
-            ", RECEIVE TICKET: " + messagereceived.ticket +
-            ", RECEIVE ROLES: " + messagereceived.roles +
-            ", RECEIVE DATA: " + messagereceived.data);
+      Message receivedMessage = deserializeMessage(msg);
+      if (receivedMessage.op != OP.PING) {
+        LOG.debug("RECEIVE: " + receivedMessage.op +
```
Comment (Reamer) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:255]: Please avoid the superfluous string concatenation and use this [method instead](http://www.slf4j.org/apidocs/org/slf4j/Logger.html#debug(java.lang.String,java.lang.Object...)).

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 2]
```diff
@@ -251,27 +250,27 @@ public void onOpen(NotebookSocket conn) {
...
+            ", RECEIVE PRINCIPAL: " + receivedMessage.principal +
+            ", RECEIVE TICKET: " + receivedMessage.ticket +
+            ", RECEIVE ROLES: " + receivedMessage.roles +
+            ", RECEIVE DATA: " + receivedMessage.data);
       }
       if (LOG.isTraceEnabled()) {
```
Comment (Reamer) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:261]: If you use the other LOG method, this `if` is unnecessary.

**Review by Reamer — APPROVED** [review]
Just a small note. Great work.

**Comment by zjffdu** [comment]
ping @cuspymd There're some comments unresolved

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java (hunk 1 above)
Comment (cuspymd) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:255]: This PR is delayed too much, so I will fix it with a separate PR later.

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java (hunk 2 above)
Comment (cuspymd) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java:261]: This PR is delayed too much, so I will fix it with a separate PR later.

**Pull request closed without merging**